webapp/app/forms/steps/lotse/personal_data.py

```diff
@@ -100,6 +100,16 @@ class InputForm(SteuerlotseBaseForm):
             label=_l('form.lotse.steuernummer.request_new_tax_number'),
             render_kw={'data_label': _l('form.lotse.steuernummer.request_new_tax_number.data_label')})
 
+        def __init__(self, *args, **kwargs):
+            super().__init__(*args, **kwargs)
+            tax_offices = request_tax_offices()
```

Comment:
I am not saying the other way is better but why did you decide to call `request_tax_offices` here instead of overwriting `prepare_render_info`?

Reply:
I am wondering if we should add tests for this? Apparently we only caught it with the acceptance tests?

Reply:
tests will be added in a separate PR

Reply:
Overwriting `prepare_render_info` requires much more changes on different inheritance levels of a step. I tried it and it got messy pretty quickly. E.g. this is the method on the FormSteuerlotseStep:

```
    def prepare_render_info(cls, stored_data, input_data=None, should_update_data=False, *args, **kwargs):
        render_info = super().prepare_render_info(stored_data, input_data, should_update_data, *args, **kwargs)

        # IDEALLY, THIS SHOULD GO HERE

        render_info.form = cls.create_form(form_data=input_data, prefilled_data=stored_data)

        if should_update_data and render_info.form.validate():
            render_info.data_is_valid = True
            stored_data.update(render_info.form.data)

        render_info.stored_data = stored_data

        return render_info
``` 
The ideal place is the marked one (after constructing the render_info to be able to set it there and before constructing + validating the form to validate correctly) - this is tricky if we still want to call `super()` which we probably want to.